transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -289,15 +288,23 @@ public boolean isActive() {
      */
     @SuppressWarnings("unchecked")
     public <T> T readInbound() {
-        return (T) poll(inboundMessages);
+        T message = (T) poll(inboundMessages);
+        if (message != null) {
+            ReferenceCountUtil.touch(message, "Caller of readInbound() will handle the message from this point");
```

Comment:
The message is inspired by https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java#L1021.

Reply:
ReferenceCountUtil.touch handles null I think.